### PR TITLE
[FEATURE] Refactor SwapAxis operator.

### PR DESCRIPTION
**Pull request by agrabows**
## Description ##
SwapAxis operator was implemented without use of NNVM_REGISTER_OP. Refactoring with new API would enable integrating oneDNN's reorder primitive support which would speed up gpt2 model.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage

### Changes ###
- [x] Refactor SwapAxis operator with NNVM_REGISTER_OP.

### CODE REVIEW

File: src/operator/swapaxis-inl.h [hunk 1]
```diff
@@ -53,223 +53,157 @@ struct SwapAxisParam : public dmlc::Parameter<SwapAxisParam> {
...
+              const OpContext& ctx,
+              const std::vector<TBlob>& in_data,
+              const std::vector<TBlob>& out_data,
+              const std::vector<OpReqType>& req) {
+  using namespace mshadow;
+  using namespace mshadow::expr;
+
+  TBlob data_in              = in_data[swapaxisenum::kData];
+  TBlob data_out             = out_data[swapaxisenum::kOut];
+  OpReqType out_req          = req[swapaxisenum::kData];
```
[the remark this answers is not in the document]
Comment (agrabows) [reply src/operator/swapaxis-inl.h]: done

**Comment by agrabows** [comment]
@mxnet-bot run ci [unix-cpu, website]

**Comment by agrabows** [comment]
@mxnet-bot run ci [website]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [website]

**Comment by agrabows** [comment]
@mxnet-bot run ci [centos-gpu, edge, website]

**Comment by agrabows** [comment]
@mxnet-bot run ci [website, centos-gpu]

**Comment by agrabows** [comment]
@mxnet-bot run ci [website, centos-gpu]

**Comment by agrabows** [comment]
@mxnet-bot run ci [centos-gpu, windows-gpu, unix-cpu, website]

**Comment by agrabows** [comment]
@mxnet-bot run ci [centos-gpu, windows-gpu, unix-cpu, website]

**Comment by agrabows** [comment]
@mxnet-bot run ci [unix-cpu]

**Review by anko-intel — APPROVED** [review]

**Pull request merged**